packages/flutter/lib/src/gestures/long_press.dart

```diff
@@ -133,89 +133,171 @@ class LongPressGestureRecognizer extends PrimaryPointerGestureRecognizer {
+  ///    callback.
   GestureLongPressEndCallback onLongPressEnd;
 
+  @override
+  bool isPointerAllowed(PointerDownEvent event) {
+    switch (event.buttons) {
+      case kPrimaryButton:
+        if (onLongPressStart == null &&
+            onLongPress == null &&
+            onLongPressMoveUpdate == null &&
+            onLongPressEnd == null &&
+            onLongPressUp == null)
+          return false;
+        break;
+      default:
+        return false;
+    }
+    return super.isPointerAllowed(event);
+  }
+
   @override
   void didExceedDeadline() {
+    // Exceeding the deadline puts the gesture in the accepted state.
     resolve(GestureDisposition.accepted);
     _longPressAccepted = true;
     super.acceptGesture(primaryPointer);
-    if (onLongPress != null) {
-      invokeCallback<void>('onLongPress', onLongPress);
-    }
-    if (onLongPressStart != null) {
-      invokeCallback<void>('onLongPressStart', () {
-        onLongPressStart(LongPressStartDetails(
-          globalPosition: _longPressOrigin,
-        ));
-      });
-    }
+    _checkLongPressStart();
   }
 
   @override
   void handlePrimaryPointer(PointerEvent event) {
     if (event is PointerUpEvent) {
       if (_longPressAccepted == true) {
-        if (onLongPressUp != null) {
-          invokeCallback<void>('onLongPressUp', onLongPressUp);
-        }
-        if (onLongPressEnd != null) {
-          invokeCallback<void>('onLongPressEnd', () {
-            onLongPressEnd(LongPressEndDetails(
-              globalPosition: event.position,
-            ));
-          });
-        }
-        _longPressAccepted = false;
+        _checkLongPressEnd(event);
       } else {
+        // Pointer is lifted before timeout.
         resolve(GestureDisposition.rejected);
       }
-    } else if (event is PointerDownEvent || event is PointerCancelEvent) {
+      _reset();
+    } else if (event is PointerCancelEvent) {
+      _reset();
+    } else if (event is PointerDownEvent) {
       // The first touch.
-      _longPressAccepted = false;
       _longPressOrigin = event.position;
-    } else if (event is PointerMoveEvent && _longPressAccepted && onLongPressMoveUpdate != null) {
-      invokeCallback<void>('onLongPressMoveUpdate', () {
-        onLongPressMoveUpdate(LongPressMoveUpdateDetails(
-          globalPosition: event.position,
-          offsetFromOrigin: event.position - _longPressOrigin,
-        ));
-      });
+      _initialButtons = event.buttons;
+    } else if (event is PointerMoveEvent) {
+      if (event.buttons != _initialButtons) {
+        resolve(GestureDisposition.rejected);
+        stopTrackingPointer(primaryPointer);
+      } else if (_longPressAccepted) {
+        _checkLongPressMoveUpdate(event);
+      }
+    }
+  }
+
+  void _checkLongPressStart() {
+    final LongPressStartDetails details = LongPressStartDetails(
+      globalPosition: _longPressOrigin,
+    );
+    switch (_initialButtons) {
+      case kPrimaryButton:
```

Comment:
can it be anything else at this point?

Reply:
You're right. I've removed all similar checks in https://github.com/flutter/flutter/pull/30339/commits/e8835c9b5f1a719e7f448dfbf87471ad576fcf28